main_objs.py

```diff
@@ -220,6 +240,22 @@ def getBeatStrength(self):
             self.analyses['BeatStrength'] = df
         return self.analyses['BeatStrength']
 
+    def getTimeSignature(self):
+        """
+        Return a data frame containing the time signatures and their offsets
+        """
+
+        if 'TimeSignature' not in self.analyses:
+            time_signatures = []
+            for part in self._getFlatParts():
+                time_signatures.append(pd.Series({ts.offset: ts for ts in part.getTimeSignatures()}))
+            # immediate = self._getFlatParts().apply(lambda part:part.getTimeSignatures)
+            # ans = immediate.apply(lambda ts: ts.offset)
+            df = pd.concat(time_signatures, axis=1)
+            df.columns = self._getPartNames()
+            self.analyses['TimeSignature'] = df
```

Comment:
You're returning a df of music21 time signature objects, and users won't know what to do with those. music21 time signature objects have a .ratioString property for this. We need to use applymap to apply a function to each cell. You can use the `na_action` parameter of .applymap to ignore the NaN values so you don't have to handle them (they just propagated). So:
df = df.applymap(lambda ts: ts.ratioString, na_action='ignore')